pkg/inject/inject.go

```diff
@@ -0,0 +1,541 @@
+package inject
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/linkerd/linkerd2/controller/gen/config"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	log "github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	k8sMeta "k8s.io/apimachinery/pkg/api/meta"
+	k8sResource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// LocalhostDNSNameOverride allows override of the controlPlaneDNS. This
+	// must be in absolute form for the proxy to special-case it.
+	LocalhostDNSNameOverride = "localhost."
+	// ControlPlanePodName default control plane pod name.
+	ControlPlanePodName = "linkerd-controller"
+	// PodNamespaceEnvVarName is the name of the variable used to pass the pod's namespace.
+	PodNamespaceEnvVarName = "LINKERD2_PROXY_POD_NAMESPACE"
+	// PrometheusImage is the docker image and tag for the Prometheus instance used in the control plane
+	PrometheusImage                 = "prom/prometheus:v2.7.1"
+	prometheusProxyOutboundCapacity = 10000
+
+	defaultDockerRegistry = "gcr.io/linkerd-io"
```

Comment:
mind sharing this with `defaultDockerRegistry` in `root.go`? same with `defaultKeepaliveMs`.